src/org/chartjs/HorizontalBarDAOChartView.js

```diff
@@ -0,0 +1,100 @@
+foam.CLASS({
+  package: 'org.chartjs',
+  name: 'HorizontalBarDAOChartView',
+  extends: 'org.chartjs.AbstractChartView',
+
+  implements: [
+    'foam.mlang.Expressions'
+  ],
+
+  properties: [
+    {
+      class: 'Map',
+      name: 'config',
+      documentation: `
+        The config map that is expected by chartjs. Structure and information can be found in chartjs.org's documentation.
+      `,
+      factory: function () {
+        return {
+          type: 'horizontalBar',
+        };
+      }
+    },
+    {
+      class: 'Map',
+      name: 'customDatasetStyling',
+      documentation: `
+        Property map that would hold the customization for each key type
+        1. Key must equal the candlestick's key.
+        2. Value mapped with key must be a 1:1 mapping defined in chartjs.org's documentation.
+      `
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'keyExpr',
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'xExpr',
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'yExpr',
+    }
+  ],
+
+  reactions: [
+    ['', 'propertyChange.yExpr', 'dataUpdate'],
+    ['', 'propertyChange.data', 'dataUpdate']
+  ],
+
+  listeners: [
+    {
+      name: 'dataUpdate',
+      isFramed: true,
+      code: function () {
+        var self = this;
+        self.data
+          .orderBy(this.yExpr)
+          .select(this.GROUP_BY(this.yExpr, this.GROUP_BY(this.keyExpr, this.SUM(this.xExpr))))
+          .then(function (sink) {
+            // Clear data before cloning because it gets clobbered anyway.
+            self.config.data = { datasets: [] };
```

Comment:
Do we really want to clear the datasets and then set them again?  Will that trigger multiple plains from the Charts?  We could also just build the new datasets and just do self.config.data = ..., followed by i believe an .update() call on the chartjs object.

I guess this works it just seems like unnecessary cloning.

Reply:
Maybe? I remember having issues with changing just the datasets on chartjs so we clone the config and re-render the entire chart.

Leave this for now if it works and I'll think about doing it better.